### DISCUSSION TITLE
[1.16.4 - 1.94.0] Pcall Throws Error if Function Takes too Long

### DISCUSSION THREAD

**Discussion by Flexico** [cc-tweaked/CC-Tweaked#605] [Q&A] [answered]
![2020-12-12_15 33 08](https://user-images.githubusercontent.com/5751770/101994455-96ecf180-3c90-11eb-8429-d0ce4fbabb5e.png)
![2020-12-12_15 33 33](https://user-images.githubusercontent.com/5751770/101994458-9a807880-3c90-11eb-9900-3c1265b98a3c.png)

It seems to stop and throw the error after about 7 seconds of calculating, and the line specified seems to be a random spot within the recursion block.

**Comment by Lemmmy** [comment, marked as answer]
This is not a bug. You are required to yield at least once every 5 seconds, or your program will be terminated. Consider inserting `sleep(0)` at the end of your loop (you can only draw once per tick anyway).

Tip: for future reference, you can ask programming related questions in the new [Discussions](https://github.com/SquidDev-CC/CC-Tweaked/discussions) page.

**Comment by SquidDev** [comment]
As Lem mentioned, this is intentional behaviour. We don't want badly written programs to end up using excessive resources, and so there are some requirements on how long you can execute for.

Any yield (`os.pullEvent()`, `sleep`, etc...) will reset the timer, so inserting one of those should resolve the issue.

**Comment by Flexico** [comment]
Ah, thanks for the info! One question though: Are you sure it can only draw once per tick? Because so far I've been drawing hundreds of pixels per second, not 20!

**Comment by Lemmmy** [comment]
The monitor only refreshes once per tick, is what I meant

**Reply by Flexico** [reply]
Ah ok, running "sleep(0)" every loop slowed things WAY down, but a timer that only fires it off every 4 seconds seems to do the trick! Thanks a lot guys~
